### PR TITLE
[Bug-fix] Fix Be cores when user specified HLL or BITMAP as operand of BinaryPredicate

### LINKED ISSUE #5798: [BUG] Be cores when user specified `HLL` or `BITMAP` as operand of BinaryPredicate
**Describe the bug**
Be cores when user specified HLL or BITMAP type as operand of BinaryPredicate

**To Reproduce**
1. create a table with `hll` `column
```
CREATE TABLE `test_uv` (
  `dt` date NULL COMMENT "",
  `uv_set` hll HLL_UNION NULL COMMENT ""
) ENGINE=OLAP
AGGREGATE KEY(`dt`)
DISTRIBUTED BY HASH(`dt`) BUCKETS 32
PROPERTIES (
"replication_num" = "1",
);
```
2. insert some data
```
insert into test_uv values ('2021-05-12', hll_hash('YEUSKDF'));
```

3.select
```
select * from test_uv where uv_set = 'ONE STRING' limit 1;
```

**Result**
Be cores

### PR DESCRIPTION
**Pull request by xy720**
## Proposed changes

Fix be cores when user specified HLL or BITMAP type as operand of BinaryPredicate.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #5798 ) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/test/java/org/apache/doris/analysis/BinaryPredicateTest.java [hunk 1]
```diff
@@ -17,20 +17,19 @@
 
 package org.apache.doris.analysis;
 
+import mockit.Expectations;
```
Comment (EmmyMiao87) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/BinaryPredicateTest.java:20]: Please keep the order of import package

File: fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java [hunk 2]
```diff
@@ -278,7 +278,16 @@ private boolean canCompareDate(PrimitiveType t1, PrimitiveType t2) {
...
+            if (e.getType().getPrimitiveType() == PrimitiveType.HLL) {
+                throw new AnalysisException("Hll type can not be a operand: " + toSql());
+            }
+            if (e.getType().getPrimitiveType() == PrimitiveType.BITMAP) {
+                throw new AnalysisException("Bitmap type can not be a operand: " + toSql());
+            }
+        }
+
         PrimitiveType t1 = getChild(0).getType().getResultType().getPrimitiveType();
         PrimitiveType t2 = getChild(1).getType().getResultType().getPrimitiveType();
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java:293]: Please put in here

**Review by EmmyMiao87 — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java [hunk 3]
```diff
@@ -278,10 +278,19 @@ private boolean canCompareDate(PrimitiveType t1, PrimitiveType t2) {
...
     }
 
-    private Type getCmpType() {
+    private Type getCmpType() throws AnalysisException {
         PrimitiveType t1 = getChild(0).getType().getResultType().getPrimitiveType();
         PrimitiveType t2 = getChild(1).getType().getResultType().getPrimitiveType();
 
+        for (Expr e : getChildren()) {
+            if (e.getType().getPrimitiveType() == PrimitiveType.HLL) {
+                throw new AnalysisException("Hll type can not be a operand: " + toSql());
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java:287]:
```suggestion
                throw new AnalysisException("Hll type does not support operand: " + toSql());
```

File: fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java [hunk 4]
```diff
@@ -278,10 +278,19 @@ private boolean canCompareDate(PrimitiveType t1, PrimitiveType t2) {
...
+            }
+            if (e.getType().getPrimitiveType() == PrimitiveType.BITMAP) {
+                throw new AnalysisException("Bitmap type can not be a operand: " + toSql());
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java:290]:
```suggestion
                throw new AnalysisException("Bitmap type does not support operand: " + toSql());
```

File: fe/fe-core/src/test/java/org/apache/doris/analysis/BinaryPredicateTest.java (hunk 1 above)
Comment (xy720) [reply fe/fe-core/src/test/java/org/apache/doris/analysis/BinaryPredicateTest.java]: ok

File: fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java (hunk 2 above)
Comment (xy720) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/BinaryPredicate.java]: ok

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**